### DISCUSSION TITLE
physics vehicle

### DISCUSSION THREAD

**Discussion by WOLFxxxxxx** [bjornbytes/lovr#781] [General]
Is it possible to create a physics vehicle
![forces-moving-car](https://github.com/bjornbytes/lovr/assets/128394802/2bc7e5b9-d293-4385-b70e-42f58c3c28b8)
?

**Comment by jmiskovic** [comment]
The included illustration shows a simple physical model that is very much possible in LOVR, both with the ODE engine in stable version and in Jolt engine that the LOVR is switching to. The drag, weight and reactive force is automatically handled by the physics engine, just set the mass and linear/angular damping for the collider. For LOVR+ODE, there is (a bit dated) example [here](https://github.com/jmiskovic/moon-safari/blob/main/Vehicle.lua) with online demo.

A more fun physical model is described in [this video](https://www.youtube.com/watch?v=LG1CtlFRmpU). It's more complex, requiring you to raycast downwards, and calculate friction depending on vehicle orientation and direction. I've implemented most of ideas from the video and I can help if you run into issues.

Then there are even more complex models like one described in article [Car Physics for Games](https://www.asawicki.info/Mirror/Car%20Physics%20for%20Games/Car%20Physics%20for%20Games.html), with modeling for wheel suspensions, slipping and transmission gears. Jolt has some vehicle helpers built-in that reference that article. Even though the LOVR doesn't expose those helpers in physics API, all the same concepts can be implemented on Lua side of things.

**Reply by WOLFxxxxxx** [reply]
thank you for help💜
